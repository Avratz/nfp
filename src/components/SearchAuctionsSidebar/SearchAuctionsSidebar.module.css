.searchAuctionsSidebar{
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  margin: 1.5rem 0;
}

.divider{
  width: 100%;
  height: 1px;
  background-color: var(--neutral-3);
}

.resetButton{
  display: flex;
  align-items: center;
  gap: 0.5rem;
  font-size: var(--text-sm);
  line-height: var(--line-sm);
}