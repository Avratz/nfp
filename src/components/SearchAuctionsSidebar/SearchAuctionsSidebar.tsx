import { Button } from '@/ui/Button/Button'
import { Icon } from '@/ui/Icon/Icon'
import { Range } from '@/ui/Range/Range'
import { SelectBox } from '@/ui/SelectBox/SelectBox'
import { cx } from 'class-variance-authority'

import styles from './SearchAuctionsSidebar.module.css'

const colors = [
  { name: 'Red', value: 'red' },
  { name: 'Blue', value: 'blue' },
  { name: 'Green', value: 'green' },
]

export function SearchAuctionsSidebar() {
  return (
    <aside className={styles.searchAuctionsSidebar}>
      <Range min={0.01} max={10} step={0.01} unit="ETH" label="Price Range" />
      <div className={styles.divider}></div>
      <SelectBox
        placeholder="Most liked"
        setSelected={(value) => null}
        label="Likes"
      >
        <SelectBox.Option value="most-liked">Most Liked</SelectBox.Option>
        <SelectBox.Option value="least-liked">Least Liked</SelectBox.Option>
      </SelectBox>
      <SelectBox
        placeholder="Colors"
        setSelected={(value) => null}
        label="Open"
      >
        {colors.map((color) => (
          <SelectBox.Option key={color.name} value={color}>
            <span className={cx(styles.dot, styles[color.name])}></span>
            {color.name}
          </SelectBox.Option>
        ))}
      </SelectBox>
      <div className={styles.divider}></div>
      <Button intent="blank" className={styles.resetButton}>
        <Icon name="closeCircle" />
        <span>Reset filter</span>
      </Button>
    </aside>
  )
}
