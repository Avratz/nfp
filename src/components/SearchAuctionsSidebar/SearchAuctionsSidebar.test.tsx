import { render } from '@testing-library/react'
import { SearchAuctionsSidebar } from './SearchAuctionsSidebar'

jest.mock('@/ui/Icon/Icon', () => {
  function Icon() {
    return <div></div>
  }

  Icon.displayName = 'Icon'
  return {
    __esModule: true,
    Icon,
  }
})

describe('SearchAuctionsSidebar', () => {
  it('renders SearchAuctionsSidebar', () => {
    let { container } = render(<SearchAuctionsSidebar />)
    expect(container).toMatchSnapshot()
  })
})
