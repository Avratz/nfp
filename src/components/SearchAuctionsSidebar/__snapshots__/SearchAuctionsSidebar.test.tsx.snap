// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAuctionsSidebar renders SearchAuctionsSidebar 1`] = `
<div>
  <aside
    class="searchAuctionsSidebar"
  >
    <label
      class="rangeContainer"
    >
      <div
        class="label"
      >
        Price Range
      </div>
      <div
        class="rangeBox"
      >
        <input
          class="inputRange"
          max="10.05"
          min="0.001"
          step="0.001"
          type="range"
          value="10.05"
        />
        <div
          class="track"
        />
        <div
          class="progress"
        />
        <div
          class="dot"
          style="left: 100%; transform: translateX(-100%);"
        >
          <div
            class="tooltip"
          >
            10.05
             ETH
          </div>
        </div>
      </div>
      <datalist
        class="units"
      >
        <option
          label="0.001 ETH"
        />
        <option
          label="10.05 ETH"
        />
      </datalist>
    </label>
    <hr
      class="divider horizontal"
    />
    <div
      class="selectBox"
    >
      <label
        class="label"
        data-headlessui-state=""
        id="headlessui-listbox-label-:r0:"
      >
        Likes
      </label>
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="headlessui-listbox-label-:r0: headlessui-listbox-button-:r1:"
        class="button"
        data-headlessui-state=""
        id="headlessui-listbox-button-:r1:"
        type="button"
      >
        <span>
          Most Liked
        </span>
        <span
          class="icon"
        >
          <div />
        </span>
      </button>
    </div>
    <div
      class="selectBox"
    >
      <label
        class="label"
        data-headlessui-state=""
        id="headlessui-listbox-label-:r3:"
      >
        Colors
      </label>
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="headlessui-listbox-label-:r3: headlessui-listbox-button-:r4:"
        class="button"
        data-headlessui-state=""
        id="headlessui-listbox-button-:r4:"
        type="button"
      >
        <span>
          All colors
        </span>
        <span
          class="icon"
        >
          <div />
        </span>
      </button>
    </div>
    <hr
      class="divider horizontal"
    />
    <button
      class="button blank base resetButton"
    >
      <div />
      <span>
        Reset filter
      </span>
    </button>
  </aside>
</div>
`;
